releng/jenkins/release-prepare-branches/Jenkinsfile

```diff
@@ -0,0 +1,236 @@
+    GITHUB_API_CREDENTIALS_ID = 'github-bot-token'
+  }
+
+  stages {
+    stage('Prepare') {
+      steps {
+        // checkout xtext-build-tools
+        checkout scm
+        
+        script {
+          env.xtextVersion = getCurrentXtextVersion(params.SOURCE_BRANCH)
+          // for stable releases append the release qualifier
+          if (params.RELEASE != 'GA') {
+            env.xtextVersion += "."+params.RELEASE
+          }
+          env.tagName="v${xtextVersion}"
+          env.branchName="${params.RELEASE != 'GA' ? 'milestone_'+xtextVersion : 'release_'+xtextVersion}"
+
+          if (!xtextVersion.startsWith('2.')) {
+            currentBuild.result = 'ABORTED'
+            error('XTEXT_VERSION invalid')
+          }
+          currentBuild.displayName = "#${env.BUILD_NUMBER} ${xtextVersion}";
+          if (params.DRY_RUN) {
+            currentBuild.displayName += " (Dry Run)"
+          }
+
+          println "xtext version to be released ${xtextVersion}"
+          println "branch to be created ${branchName}"
+          println "tag to be created ${tagName}"
+
+          sshagent([CREDENTIAL_ID_GENIE_XTEXT_GITHUB]) {
+            wrap([$class: 'BuildUser']) {
+              repositoryNames.split(',').each {
+                sh """
+                  git clone -b ${params.SOURCE_BRANCH} --depth 1 ${baseGitURL}/${it}.git
+                  cd ${it}
+                  git config user.name "${env.BUILD_USER}"
+                  git config user.email "${env.BUILD_USER_EMAIL}"
+                  git checkout -b ${branchName}
+                  cd ..
+                """
+              }
+            }
+          } // END sshagent
+        } // END script
+      } // END steps
+    } // END stage
+    
+    stage('Modify') {
+      steps {
+        script {
+          repositoryNames.split(',').each {
+            print "##### Preparing $it ########"
+            dir(it) {
+              modifyFiles (it, xtextVersion, branchName)
+            }
+          }
+        } // END script
+      } // END steps
+    } // END stage
+
+
+    stage('Commit & Push') {
+      steps {
+        script {
+          repositoryNames.split(',').each {
+            dir (it) {
+              // Avoid commit when no change has happened. This would make the pipeline fail.
+              sh label: "Commit changes", script: """
+                echo "#### Commit changes for ${it} ####"
+                git diff-index --quiet HEAD || git commit --signoff -a --allow-empty -m '[release] version $xtextVersion'
+                git tag --force -a ${tagName} -m 'release ${tagName}'
+              """
+              if (params.DRY_RUN || params.VERBOSE) {
+                sh "git diff ${params.SOURCE_BRANCH}..${branchName}"
+              }
+            }
+          }
+          if (!params.DRY_RUN) {
+            withCredentials([string(credentialsId: "${GITHUB_API_CREDENTIALS_ID}", variable: 'GITHUB_API_TOKEN')]) {
```

Comment:
do we really need api credentials AND the ssh agents? `git push` looks like the ssh agent should be enough?

Reply:
I'll give it a try. I assume you found a small issue here - nice catch. When the branches are pushed without API credentials, I'll remove that.

Reply:
Works.